agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -143,6 +146,19 @@ func (state *DockerTaskEngineState) ContainerMapByArn(arn string) (map[string]*a
 	return ret, ok
 }
 
+// TaskByShortID retrieves the task of a given docker short container id
+func (state *DockerTaskEngineState) TaskByShortID(cid string) (*api.Task, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	for id := range state.idToTask {
+		if strings.HasPrefix(id, cid) {
```

Comment:
I think it would be reasonable to have `func (state *DockerTaskEngineState) TaskByShortID(cid string) ([]*api.Task, bool)`.